services/storage/src/simcore_service_storage/api/_worker_tasks/_simcore_s3.py

```diff
@@ -53,3 +60,44 @@ async def deep_copy_files_from_project(
             )
 
     return body.destination
+
+
+async def data_export(
+    task: Task,
+    task_id: TaskID,
+    *,
+    user_id: UserID,
+    paths_to_export: list[S3ObjectKey],
+) -> StorageFileID:
+    _logger.info("Exporting (for user='%s') selection: %s", user_id, paths_to_export)
+
+    dsm = get_dsm_provider(get_fastapi_app(task.app)).get(
+        SimcoreS3DataManager.get_location_id()
+    )
+    assert isinstance(dsm, SimcoreS3DataManager)  # nosec
+
+    try:
+        for path_to_export in paths_to_export:
+            await dsm.can_read_file(user_id=user_id, file_id=path_to_export)
+    except FileAccessRightError as err:
+        raise AccessRightError(
+            user_id=user_id,
+            file_id=path_to_export,
+            location_id=SimcoreS3DataManager.get_location_id(),
+        ) from err
+
+    with get_tqdm_progress(total=1, description=f"{task.name}") as pbar:
```

Comment:
I don't think this is making your code clearer here.

The ProgressBarData is already calling the callback above periodically.
Suggestion:
1. remove tqdm to not have 2 different progress bars, that make the code confusing (why 2 progress??)
2. use the callback function if you want to print in the log instead.